### PR TITLE
Closes #621 : Upgrade to SB 3.2.x and Wildfly 31.0.1.Final

### LINKED ISSUE #621: Upgrade to Spring Boot 3.2
Upgrade to SpringBoot 3.2

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.2-Release-Notes

Refactore RestTemplate to use new RestClient

https://docs.spring.io/spring-boot/docs/3.2.0-SNAPSHOT/reference/html//io.html#io.rest-client.restclient

### PR DESCRIPTION
**Pull request by arolfes**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review

### Verified by the reviewer:
- [ ] Commit message format → ADPT-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml [hunk 1]
```diff
@@ -92,6 +91,9 @@
       <logger category="com.arjuna">
         <level name="WARN"/>
       </logger>
+      <logger category="com.networknt.schema">
```
Comment (ryzheboka) [inline taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml:94]: Why did you add this new logger?

File: taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml [hunk 2]
```diff
@@ -437,15 +431,24 @@
...
           <permission class-name="org.wildfly.transaction.client.RemoteTransactionPermission"
                       module="org.wildfly.transaction.client"/>
           <permission class-name="org.jboss.ejb.client.RemoteEJBPermission"
                       module="org.jboss.ejb-client"/>
+          <permission class-name="org.wildfly.extension.batch.jberet.deployment.BatchPermission"
+                      module="org.wildfly.extension.batch.jberet" target-name="*"/>
         </permission-set>
       </permission-sets>
       <http>
+        <http-authentication-factory name="application-http-authentication"
```
Comment (ryzheboka) [inline taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml:443]: Why did you add this new http-authenrication-factory and the new security-domain?

**Review by ryzheboka — COMMENTED** [review]
Thanks for the PR! Could you explain some changes to the int-test-standalone.xml? I couldn't understand them myself. Especially the new usage of "ApplicationDomain" and "ManagementDomain"

**Comment by arolfes** [comment]
Hello @ryzheboka ,

> Thanks for the PR! Could you explain some changes to the int-test-standalone.xml? I couldn't understand them myself. Especially the new usage of "ApplicationDomain" and "ManagementDomain"

i downloaded the Version 31.0.1.Final from Wildfly manually. Then I compared the original config file with our config file. Everything new I added to our config file.

Please see attachment. [standalone-full.xml.txt](https://github.com/Taskana/TaskanaAdapter/files/15033516/standalone-full.xml.txt)

File: taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml (hunk 2 above)
Comment (arolfes) [reply taskana-adapter-camunda-wildfly-example/src/test/resources/int-test-standalone.xml:443]: I copied it from original config file
